CMakeLists.txt

```diff
@@ -14,7 +14,7 @@ endif()
 
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 
-option(QT_DEFAULT_MAJOR_VERSION "Major version of Qt to use (6 or 5)" 6)
+set(VCPKG_QT_DEFAULT_MAJOR_VERSION 6 CACHE STRING "Major version of Qt to use (6 or 5)")
```

Comment:
Is there any reason why this is not an `option` and instead a `set` ? 

Also if this is now a VCPKG only option, should it be defined inside the `if(VCPKG)` below ?

Reply:
If I understand cmake correctly, `option(NAME ...)` is equivalent to `set(NAME ... CACHE BOOL ...)`, and `5|6` isn't bool.

> Also if this is now a VCPKG only option, should it be defined inside the if(VCPKG) below ?

I didn't try that because I'm not sure if config knob can be conditionally defined. Anyway, it's probably better to have a common user variable to select Qt version.